packages/core/src/lib/assets/createImagesContext.ts

```diff
@@ -0,0 +1,132 @@
+import path from 'path';
+import { v4 as uuid } from 'uuid';
+import fs from 'fs-extra';
+import fromBuffer from 'image-type';
+import imageSize from 'image-size';
+import { KeystoneConfig, ImageMetadata, ImagesContext } from '../../types';
+import { AssetsAPI } from './types';
+const DEFAULT_BASE_URL = '/images';
+export const DEFAULT_IMAGES_STORAGE_PATH = './public/images';
+
+export function getImageMetadataFromBuffer(buffer: Buffer): ImageMetadata {
+  const fileType = fromBuffer(buffer);
+  if (!fileType) {
+    throw new Error('File type not found');
+  }
+
+  const extension = fileType.ext;
+  if (extension !== 'jpg' && extension !== 'png' && extension !== 'webp' && extension !== 'gif') {
+    throw new Error(`${extension} is not a supported image type`);
+  }
+
+  const { height, width } = imageSize(buffer);
+
+  if (width === undefined || height === undefined) {
+    throw new Error('Height and width could not be found for image');
+  }
+  return { width, height, filesize: buffer.length, extension };
+}
+
+export function createImagesContext(
+  config: KeystoneConfig,
+  s3Assets: () => Map<string, AssetsAPI>
```

Comment:
The whole function thing just existed because the cloud stuff had to do async setup that wouldn't necessarily have been done by the time this was called. Could you just pass it directly?

Reply:
I still like having this for creating the local files, and for making this relatively long code block have its own file

Reply:
Oh, yeah, I don't mean moving anything, just changing `s3Assets: () => Map<string, AssetsAPI>` to `s3Assets: Map<string, AssetsAPI>`